client/src/utilities/solvingAlgorithms.ts

```diff
@@ -0,0 +1,233 @@
+import { Point } from "./types";
+
+export const dfs = (
+    maze: number[][],
+    curr: Point | null,
+    start: Point | null,
+    end: Point | null,
+    seen: boolean[][],
+    path: Point[],
+    setMaze: React.Dispatch<React.SetStateAction<number[][]>>,
+    delay: number,
+    setSolving: React.Dispatch<React.SetStateAction<boolean>>,
+    solvingRef: React.MutableRefObject<boolean>
+): Promise<boolean> => {
+    return new Promise((resolve) => {
```

Comment:
Recursing with promises is not a good idea because it can lead to a large number of pending promises and increased complexity in managing the asynchronous control flow.